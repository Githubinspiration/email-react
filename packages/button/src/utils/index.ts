export * from "./px-to-pt";
