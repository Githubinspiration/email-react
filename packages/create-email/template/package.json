{
  "name": "emails",
  "version": "0.0.0",
  "private": true,
  "workspaces": [
    ".react-email"
  ],
  "scripts": {
    "dev": "email dev",
    "export": "email export"
  },
  "dependencies": {
    "@react-email/components": "0.0.3",
    "react-email": "1.7.15"
  }
}
