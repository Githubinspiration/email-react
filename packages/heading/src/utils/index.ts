export * from "./as";
export * from "./spaces";
