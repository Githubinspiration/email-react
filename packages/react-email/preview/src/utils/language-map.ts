const languageMap = {
    jsx: 'React',
    markup: 'HTML' 
}

export default languageMap;