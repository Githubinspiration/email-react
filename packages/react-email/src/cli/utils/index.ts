export * from './tree';
export * from './preview';
