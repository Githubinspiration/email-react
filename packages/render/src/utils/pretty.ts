import { html } from "js-beautify";

const defaults = {
  unformatted: ["code", "pre", "em", "strong", "span"],
  indent_inner_html: true,
  indent_char: " ",
  indent_size: 2,
  sep: "\n",
};

export const pretty = (str: string, options = {}) => {
  return html(str, { ...defaults, ...options });
};
