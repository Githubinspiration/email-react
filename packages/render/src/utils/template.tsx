import * as React from 'react';

interface TemplateProps {
  firstName: string;
}

export const Template: React.VFC<Readonly<TemplateProps>> = ({ firstName }) => (
  <>
    <h1>Welcome, {firstName}!</h1>
    <p>Thanks for trying our product. We're thrilled to have you on board!</p>
  </>
);
