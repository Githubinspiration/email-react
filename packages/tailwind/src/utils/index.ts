export * from "./get-css-for-markup";
export * from "./escape-class-name";
export * from "./get-css-class-properties-map";
export * from "./css-to-jsx-style";
export * from "./minify-css";
