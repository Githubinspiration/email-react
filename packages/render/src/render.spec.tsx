import * as React from 'react';
import { render } from './index';
import { Template } from './utils/template';

describe('render', () => {
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetModules();
  });

  it('converts a React component into HTML', () => {
    const actualOutput = render(<Template firstName="Jim" />);
    expect(actualOutput).toMatchSnapshot();
  });

  it('converts a React component into PlainText', () => {
    const actualOutput = render(<Template firstName="Jim" />, {
      plainText: true,
    });
    expect(actualOutput).toMatchSnapshot();
  });
});
