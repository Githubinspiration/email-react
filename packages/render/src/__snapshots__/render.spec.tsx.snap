// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render converts a React component into HTML 1`] = `"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\"><h1>Welcome, Jim!</h1><p>Thanks for trying our product. We&#x27;re thrilled to have you on board!</p>"`;
